client/src/components/Ticketing/ticketingmanager/Doorlistpage/doorlist.tsx

```diff
@@ -82,86 +86,189 @@ const DoorList = () => {
+    const apiRef = useGridApiContext();
+
+    const handleExport = (options: any) => {
+      apiRef.current.exportDataAsCsv(options);
+    };
+
+    return (
+      <div className='container columns-3 gap-4 ml-3 mt-3 '>
+        <button
+          className='flex flex-row  text-blue-600 gap-2
+          items-center hover:text-blue-500 px-3 py-2 rounded-lg'
+          onClick={handleExport}
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
+            <path strokeLinecap="round" strokeLinejoin="round"
+              d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M9 19l3 3m0 0l3-3m-3 3V10" />
+          </svg>
+          Export
+        </button>
+      </div>
+    );
+  };
 
   const getDoorList = async (event) => {
     try {
       const token = await getAccessTokenSilently({
         audience: process.env.REACT_APP_ROOT_URL,
         scope: 'admin',
       });
-
-      const getuser = event.target.value;
-      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${getuser}`, {
+      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${event}`, {
         headers: {
-          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
         },
       });
-      const jsonData = await response.json();
-      console.log(jsonData.data);
-
-      // doorlistData.data {id: custid, name, vip, donor: donorbadge, accomodations: seatingaccom, num_tickets, checkedin, ticketno }
-      setDoorList(jsonData.data);
-      setEventName(jsonData.data[0].eventname);
-      setDate(dayMonthDate(jsonData.data[0].eventdate));
-      setTime(militaryToCivilian(jsonData.data[0].eventtime));
+      const jsonRes = await response.json();
+      const jsonData = jsonRes.data;
+      const doorListData = jsonData.map((item, index) => {
+        const row = item.row.slice(1, -1).split(',');
+        if (!row || !row[0]) { // Check if the value in column 0 is present
+          setTicketsSold(false);
+          return null; // Exit early if no tickets sold
+        }
+        setTicketsSold(true);
+        return {
+          id: index,
+          firstname: row[1],
+          lastname: row[2],
+          num_tickets: row[12],
+          // TODO: hook arrived with redeemed vaue
+          arrived: false,
+          vip: row[3] === 't',
+          donorbadge: row[4] === 't',
+          accommodations: row[5],
+        };
+      }).filter(Boolean);
+
+      setDoorList(doorListData);
+      const rowString = jsonData[0].row.slice(1, -1);
+      const rowParts = rowString.split(',');
+      const eventNameFromData = rowParts[7].replace(/"/g, '');
+      const eventDate = rowParts[9];
+      const eventDateObject = new Date(eventDate.replace(/(\d{4})(\d{2})(\d{2})/, '$1-$2-$3'));
```

Comment:
This whole date thing piqued my curiosity so I looked into it a little bit. It looks like `Date` objects are very finicky. But, it appears that if you use `/`s instead of `-`s, then we get the functionality we're looking for w/o the need to add 1 to the date. So, `$1/$2/$3`. https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off

Reply:
Now that I think about it though. Do we even need to put it in a `Date` object. Could we just skip from the `eventDate` to the `formattedEventDate`?

Reply:
i struggled with getting the date to format properly for a while. going straight to formattedEventDate was not an option because the object has to be a string. changing from '-' to '/' does not seem to affect the result either. 
i'm sure there's a more elegant way to do this. let me know if you want me to spend more time trying to find it.